include/tins/internals.h

```diff
@@ -200,6 +204,34 @@ template<>
 struct is_unsigned_integral<uint64_t> {
     static const bool value = true;
 };
+
+#if TINS_IS_CXX11
+
+// Part of C++14 standard library
+template<bool B, class T = void>
+using enable_if_t = typename std::enable_if<B,T>::type;
+
+// Template metaprogramming trait to determine if a functor can accept another parameter as an argument
+template <class T, class P, class=void>
+struct accepts_type : std::false_type { };
+
+template <class T, class P>
+struct accepts_type<T, P, enable_if_t<
+  std::is_same<  decltype(  std::declval<T>()(std::declval<P>())  ), bool>::value
```

Comment:
This trait seems to not work if your functor has this signature ```bool(Packet&)```, which is one of the compatible signatures for the functor when taking PDUs, so I think it should be extended to Packets as well. This can be fixed by using ```declval<P&>```, which seems very specific for this use, but so is this trait.